### DISCUSSION TITLE
Add a way to "lock" the possibility of using old email to verify code (with pgp key)

### DISCUSSION THREAD

**Discussion by ghost** [community/community#18167] [Code Security]
I wouls like to ask a way in our email list, to actually lock one (who will be archived, and in some time the domain released), to prevent any push of code with this email or validate any code (even with good pgp key) coming from this email (we must let it in the account to gard the "verified" badge on our signed commit).

So having a checkbox like "only use for old commit signature" or "locked email", to hide it from any settings (commit one etc) to prevent any accidental use of it in new "signature", connection, email reception etc.

**Comment by tommyp6** [comment]
I believe this would be a very good feature to lock emails which are no longer in use or inaccessible. Addresses from old organizations or education emails would easily fall in this category, as usually after a certain period they will be disabled, and you lose access.

**Reply by Davidkazin** [reply]
I figure that is what is happening.  We are in a middle of a superstition. I’m just trying to figure it out. It’s taken me a year to get this far.

**Reply by ghost** [reply]
Actually i'm pretty sure some hack can be done with people who haven't changed an old account email they have lost access (not own the domain, or (for ex student) college email as main github email, and not have enabled 2FA) enough to take the control of the account, and if they contribute to OSS software (of private one with their right), it can be a major issue since they can generate signed commit with this.

It's why a check box (when we add new email) to lock old one for any mean other than maintain verified commit on older commit be refused.

Because for organization maintaining old email of dozen of user must be a hell on earth with the current settings.
